pkg/collector/process_test.go

```diff
@@ -142,6 +142,46 @@ func TestTCPCollectingProcess_ReceiveDataRecord(t *testing.T) {
 	assert.Error(t, err)
 }
 
+func TestTCPCollectingProcess_ReceiveMultipleDataRecords(t *testing.T) {
+	input := getCollectorInput(tcpTransport, false, false)
+	// Adjust the buffer size such there are multiple reads on the connection in
+	// the collector process.
+	input.MaxBufferSize = 50
+	cp, err := InitCollectingProcess(input)
+	// Add the templates before sending data record
+	cp.addTemplate(uint32(1), uint16(256), elementsWithValueIPv4)
+	if err != nil {
+		t.Fatalf("TCP Collecting Process does not start correctly: %v", err)
+	}
+
+	go cp.Start()
+
+	// wait until collector is ready
+	waitForCollectorReady(t, cp)
+	var conn net.Conn
+	collectorAddr := cp.GetAddress()
+	go func() {
+		conn, err = net.Dial(collectorAddr.Network(), collectorAddr.String())
+		if err != nil {
+			t.Errorf("Cannot establish connection to %s", collectorAddr.String())
+		}
+		conn.Write(validDataPacket)
+		conn.Write(validDataPacket)
+	}()
+	<-cp.GetMsgChan()
+	<-cp.GetMsgChan()
```

Comment:
Like this unit test: if we are able to decode the packet without any error, we should be able to receive them correctly.

Reply:
This test is not applicable anymore. Removed it.